R/aggregate_kicking_stats.R

```diff
@@ -0,0 +1,114 @@
+#' Summarize Weekly Kicking stats
+#'
+#' @description Build columns that aggregate kicking stats at the game level.
+#'
+#' @param pbp A Data frame of NFL play-by-play data typically loaded with
+#' [load_pbp()] or [build_nflfastR_pbp()].
+#'
+#' @examples
+#'
+#' \donttest{
+#' pbp <- nflreadr::load_pbp(2021)
+#' calculate_kicking_stats(pbp)
+#' }
+#'
+#' @return a dataframe of kicking stats
+#' @seealso <https://nflreadr.nflverse.com/reference/load_player_stats.html> for the nflreadr function to download this from repo (`stat_type = "kicking"`)
+#' @export
+calculate_kicking_stats <- function(pbp){
+
+  base_kicks <- pbp %>%
+    dplyr::filter(.data$field_goal_attempt == 1 | .data$extra_point_attempt == 1) %>%
+    dplyr::select(
+      "season",
+      "week",
+      "season_type",
+      "team" = "posteam",
+      "player_name" = "kicker_player_name",
+      "player_id" = "kicker_player_id",
+      "kick_distance",
+      "field_goal_attempt",
+      "extra_point_attempt",
+      "field_goal_result",
+      "extra_point_result"
+    ) %>%
+    dplyr::group_by(.data$season,.data$week,.data$season_type,.data$team,.data$player_name,.data$player_id) %>%
```

Comment:
I wouldn’t group by name and ID here as the names sometimes change during the season or from one season to the next. Instead group by ID only and compute the name with `custom_mode()`